### PR TITLE
Enable :if and :unless options of ActiveSupport::Callbacks

### PR DESCRIPTION
**Pull request by vicvega**
Using `:if` and `:unless` options (of ActiveSupport::Callbacks) you can write something like

```
preserve :status, if: :method
```
or 

```
preserve :status, unless: -> { condition }
```

### CODE REVIEW

**Comment by pienkowb** [comment]
Hi @vicvega, thank you for the contribution! 👍 

The code looks good, let's just add a test case for this (similar to [this one](https://github.com/pienkowb/preserve/blob/14fb061bbfd7b2e08057157376c7fcaf1ebe840b/spec/preserve_spec.rb#L38-L45)).

File: spec/preserve_spec.rb [hunk 1]
```diff
@@ -44,6 +44,15 @@
     expect(json_response[:status]).to eq(nil)
   end
 
+  it 'applies callback conditional options' do
+    ParametersController.preserve(:status, only: :index, if: -> { false })
```
Comment (pienkowb) [inline spec/preserve_spec.rb]: I think we can get rid of `only: :index` here to make it more obvious that we're only testing the `if` option.

**Review by pienkowb — APPROVED** [review]
Awesome!

**Comment by pienkowb** [comment]
@vicvega This has been released in [version 2.1.0](https://rubygems.org/gems/preserve/versions/2.1.0) 🎉

**Comment by vicvega** [comment]
Very good! Thank you

**Pull request merged**